Fails to build with GCC 13 due to overloaded-virtual error in FilterEffect

carlosgcampos opened the pull request:
#### c85962a5c0e929991e5963811da957b75d1501db
<pre>
Fails to build with GCC 13 due to overloaded-virtual error in FilterEffect
<a href="https://bugs.webkit.org/show_bug.cgi?id=259729">https://bugs.webkit.org/show_bug.cgi?id=259729</a>

Reviewed by Said Abou-Hallawa.

FilterEffect.h:44:18: error: ‘virtual bool WebCore::FilterEffect::operator==(const WebCore::FilterEffect&amp;) const’ was hidden [-Werror=overloaded-virtual=]
   44 |     virtual bool operator==(const FilterEffect&amp;) const;
      |                  ^~~~~~~~
FELighting.h:42:10: note:   by ‘bool WebCore::FELighting::operator==(const WebCore::FELighting&amp;) const’
   42 |     bool operator==(const FELighting&amp;) const;
      |          ^~~~~~~~

* Source/WebCore/platform/graphics/filters/FELighting.h:

Canonical link: <a href="https://commits.webkit.org/266539@main">https://commits.webkit.org/266539@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cf6ba3e37fe28dfb05ee6492896fd6617e195f9c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14094 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14411 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14746 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15834 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13362 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14180 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16919 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14493 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16038 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14266 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14847 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11949 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16545 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12130 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12710 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19737 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13209 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12874 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16083 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13415 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11284 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12696 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3408 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17029 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13258 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/platform/graphics/filters/FELighting.h:
```diff
@@ -62,6 +62,8 @@ class FELighting : public FilterEffect {
 protected:
     FELighting(Type, const Color& lightingColor, float surfaceScale, float diffuseConstant, float specularConstant, float specularExponent, float kernelUnitLengthX, float kernelUnitLengthY, Ref<LightSource>&&);
 
+    bool operator==(const FilterEffect&) const override { RELEASE_ASSERT_NOT_REACHED(); }
```
Can you try this using statement instead:
```
    using FilterEffect::operator==;
```

carlosgcampos replied:
It works too, thanks!

carlosgcampos committed:
Fails to build with GCC 13 due to overloaded-virtual error in FilterEffect
https://bugs.webkit.org/show_bug.cgi?id=259729

Reviewed by Said Abou-Hallawa.

FilterEffect.h:44:18: error: ‘virtual bool WebCore::FilterEffect::operator==(const WebCore::FilterEffect&) const’ was hidden [-Werror=overloaded-virtual=]
   44 |     virtual bool operator==(const FilterEffect&) const;
      |                  ^~~~~~~~
FELighting.h:42:10: note:   by ‘bool WebCore::FELighting::operator==(const WebCore::FELighting&) const’
   42 |     bool operator==(const FELighting&) const;
      |          ^~~~~~~~

* Source/WebCore/platform/graphics/filters/FELighting.h:

Canonical link: https://commits.webkit.org/266539@main
